hack/version.py

```diff
@@ -31,142 +25,135 @@
-    """
-    # We're cloning the hive repo, so there's no local branch for ocm-X.Y[-mce-M.N]; but we don't want to
-    # force the user to say 'origin/ocm-X.Y[-mce-M.N]'. Accommodate...
-    parsed, remote, _, _ = parse_branch_name(branch_arg)
-    if parsed and not remote:
-        branch_arg = "origin/{}".format(branch_arg)
 
-    # This will raise an exception if there's no such commit-ish
-    commit_hash = hive_repo.rev_parse(branch_arg).hexsha
+def find_branch(hive_repo, branch_arg):
 
-    # We need to know where master is, even if we're processing a different branch. However, if
-    # we've cloned from something other than HIVE_REPO_DEFAULT, there may not be a local `master`
-    # branch. So we may need to try origin and upstream to find it.
+    # The user may have specified a branch name for which we don't have a local ref
+    # (e.g. `mce-2.1`). Rather than forcing them to specify a qualified ref, we'll
+    # automatically look for it in the `origin` and `upstream` remotes as well.
     for remote in ("", "origin/", "upstream/"):
-        master_branch = remote + HIVE_BRANCH_DEFAULT
-        log("Trying to find master at {}".format(master_branch))
+        branch = remote + branch_arg
+        log("Trying to find {} at {}".format(branch_arg, branch))
         try:
-            master_hash = hive_repo.rev_parse(master_branch).hexsha
+            commit = hive_repo.rev_parse(branch)
         except git.BadName:
-            log("Couldn't find master at `{}`".format(master_branch))
+            log("Couldn't find {} at `{}`".format(branch_arg, branch))
         else:
-            log("Found master at `{}`".format(master_branch))
+            log("Found {} at `{}`".format(branch_arg, branch))
             break
     else:
-        log("Couldn't find master branch!")
+        log("Couldn't find {} branch!".format(branch_arg))
         sys.exit(1)
+    return commit
+
+def process_master_branch(hive_repo, branch_arg):
 
-    is_master_ancestor = is_ancestor(hive_repo, commit_hash, master_hash)
+    # This will raise an exception if there's no such commit-ish
+    commit = hive_repo.rev_parse(branch_arg)
+    master_commit = find_branch(hive_repo, HIVE_BRANCH_DEFAULT)
+
+    # This includes when the commit is equal to the master commit
+    if not hive_repo.is_ancestor(commit, master_commit):
+        log(
+            "Commit {} is not equal or and ancestor of the master branch".format(
+            commit.hexsha)
+        )
+        sys.exit(1)
+    return commit.hexsha, HIVE_VERSION_PREFIX
 
-    # Was (a commit on) an ocm-X.Y[-mce-M.N] branch specified?
-    # This will find all such branch names and add them to the channel list.
-    # However, we'll only support >1 entry in this list if we're tracking
-    # master, because otherwise which would we use to compute the semver?
-    channels = []
-    maybe_prefix = None
+
+
+def process_current_branch(hive_repo, branch_arg):
```

Comment:
It looks like this is only called in one place right now, with `HEAD` as the `branch_arg`. Given that the name of the func is already `process_current_branch`, we could probably do away with `branch_arg` and just hardcode `HEAD`.